src/proxy/BaseProxy.ts

```diff
@@ -0,0 +1,22 @@
+import {SerializationService} from '../serialization/SerializationService';
+export class BaseProxy {
+
+    protected client : any;
+    private name : string;
+    private serviceName : string;
+
+    //TODO client should be of type {real client type}
+    constructor(client : { serializationService : SerializationService}, serviceName: string, name: string) {
+        this.client = client;
+        this.name = name;
+        this.serviceName = serviceName;
+    }
+
+    protected toData(object : any) : Buffer {
```

Comment:
All proxies will extend from this class. This is to serialize the given object before sending it to Hazelcast server. This is only trivial for now because we will use JSON serialization. However this abstraction will be neccesaary when we aim for Hazelcast serialization.